openlibrary/plugins/worksearch/code.py

```diff
@@ -287,14 +287,15 @@ def do_search(
     :param sort: csv sort ordering
     :param spellcheck_count: Not really used; should probably drop
     """
+
+    if web.cookies(sfw="").sfw == 'yes':
+        fields = list(
+            WorkSearchScheme.default_fetched_fields | {'editions'} | {'subject_key'}
```

Comment:
```suggestion
            WorkSearchScheme.default_fetched_fields | {'editions'} | {'subject'}
```
The `subject_key` list contains keys that are meant to be used in URLs.  Special characters, like colons, are stripped from these values.
Use `subject` here instead of `subject_key` to get the unaltered subjects.  You'll also have to change how you're accessing the subjects in `work_search.html` after this change is made.